### PR TITLE
update coverage to version 6

### PR DESCRIPTION
**Pull request by gro1m**
Might resolve issue: https://github.com/aconrad/pycobertura/issues/133

### CODE REVIEW

File: test-requirements.txt [hunk 1]
```diff
@@ -1,3 +1,3 @@
-coverage>=4.0,<4.4
+coverage>=6.0,<6.4
 mock
 pytest-cov<2.6.0
```
Comment (aconrad) [inline test-requirements.txt:3]: I think we also need to unpin `pytest-cov`, I remember those specific versions of `pytest-cov` and `coverage` worked together but I assume we can now unpin both of them, their latest versions work together.

Comment (gro1m) [reply test-requirements.txt]: Ok, done that as well. Do you know why the coverage pipelines are now pending? :)

**Comment by gro1m** [comment]
Hi @aconrad 
Do you know why the coverage pipelines are now pending? - thanks for clarifying :)

**Comment by aconrad** [comment]
Yes, because they are failing to parse the coverage file per my comment here: https://github.com/aconrad/pycobertura/issues/133#issuecomment-1036466783

**Comment by gro1m** [comment]
ah ok...

**Comment by gro1m** [comment]
@aconrad Does this comment have any relevance for us?
https://github.com/nedbat/coveragepy/issues/578#issuecomment-413881957

**Comment by aconrad** [comment]
> @aconrad Does this comment have any relevance for us? [nedbat/coveragepy#578 (comment)](https://github.com/nedbat/coveragepy/issues/578#issuecomment-413881957)

It does, and I commented in this issue.

> Or do we really need this PR? [nedbat/coveragepy#1146](https://github.com/nedbat/coveragepy/pull/1146)

Hm, I don't know. I'm not intimately familiar with `coveragepy`. But maybe? Have you tried to checkout that branch and see if it solves the issue? If yes, we could try to push for the PR to get merged...

**Comment by gro1m** [comment]
Hi @aconrad 

> > Or do we really need this PR? [nedbat/coveragepy#1146](https://github.com/nedbat/coveragepy/pull/1146)
> 
> Hm, I don't know. I'm not intimately familiar with `coveragepy`. But maybe? Have you tried to checkout that branch and see if it solves the issue? If yes, we could try to push for the PR to get merged...

Does not change anything in my opinion, but you can check yourself with:
```
git+https://github.com/alex/coveragepy.git@patch-1 
mock
pytest-cov==3.*
```
in the `test-requirements.txt` file.
Beware though that there is a small caveat:
The `tox.ini` should actually use `--force-reinstall` for the `pip` installs, so that every time you rebuild locally the package versions are taken from the `test-requirements.txt` even if they already exist.

**Comment by gro1m** [comment]
So how can we apply https://github.com/nedbat/coveragepy/issues/578#issuecomment-413881957?
How can we switch from the `source` option to the `include` option, as I do not see yet where the `.coveragerc` comes into play in `pycobertura`...

**Comment by aconrad** [comment]
> So how can we apply https://github.com/nedbat/coveragepy/issues/578#issuecomment-413881957?
> How can we switch from the `source` option to the `include` option, as I do not see yet where the `.coveragerc` comes into play in `pycobertura`...

Unfortunately, this comment is about generating cobertura files when the code coverage occurs (during test runs). We only parse them.

**Comment by aconrad** [comment]
Hm, wait. I guess we _do_ generate our own `coverage.xml` files. We can try to change it the way they recommend it in the comment for the purpose of getting the coverage of this branch to pass in our own repository if that was your intent. At least that would be a thing to improve!

**Comment by gro1m** [comment]
Hi @aconrad Should be fixed (I tested locally with Python 3.9) with the most recent commit: https://github.com/aconrad/pycobertura/pull/135/commits/ed261ced2f74f0bd93c5d0044b2afec2e236b960. 
But I do not understand why no pipeline is getting triggered ?!

**Comment by gro1m** [comment]
And also could you maybe somehow stop the pending pipelines... :)

**Comment by aconrad** [comment]
Oh boy. Travis is saying we have insufficient credit again which is why it's not running. That's it, we're moving to GitHub Actions. I'll have to work on that in the next few days. Sorry for the inconvenience.

**Comment by aconrad** [comment]
And thanks for working on this! I don't know why .coveragerc works but not tox.ini per the documentation. Great job!

**Comment by gro1m** [comment]
> Oh boy. Travis is saying we have insufficient credit again which is why it's not running. That's it, we're moving to GitHub Actions. I'll have to work on that in the next few days. Sorry for the inconvenience.

This might also allow you to populate the Releases Tab (not only the tags) :)

**Comment by aconrad** [comment]
Sounds good. I'm very busy with work this week, not sure I'll have time to take care of this right this week.

**Comment by aconrad** [comment]
> Should be fixed (I tested locally with Python 3.9) with the most recent commit: [ed261ce](https://github.com/aconrad/pycobertura/commit/ed261ced2f74f0bd93c5d0044b2afec2e236b960).

For me, the `coverage.xml` is still generated without the `pycobertura/` prefix directory. I get the following warning:

```
venv/lib/python3.8/site-packages/coverage/inorout.py:472
  /Users/alexandre/Development/pycobertura/venv/lib/python3.8/site-packages/coverage/inorout.py:472: CoverageWarning: --include is ignored because --source is set (include-ignored)
```

Don't you get this error?

This is the same message I got when I fiddled with adding `[coverage:run]` inside `tox.ini`.

**Comment by gro1m** [comment]
> > Should be fixed (I tested locally with Python 3.9) with the most recent commit: [ed261ce](https://github.com/aconrad/pycobertura/commit/ed261ced2f74f0bd93c5d0044b2afec2e236b960).
> 
> For me, the `coverage.xml` is still generated without the `pycobertura/` prefix directory. I get the following warning:
> 
> ```
> venv/lib/python3.8/site-packages/coverage/inorout.py:472
>   /Users/alexandre/Development/pycobertura/venv/lib/python3.8/site-packages/coverage/inorout.py:472: CoverageWarning: --include is ignored because --source is set (include-ignored)
> ```
> 
> Don't you get this error?
> 
> This is the same message I got when I fiddled with adding `[coverage:run]` inside `tox.ini`.

Yes with pytest I actually do get that error message as well, but the output is still rendered correctly. Whatever that means...

**Comment by gro1m** [comment]
With `pycobertura show coverage.xml` on the other hand I do not get the error.

**Comment by gro1m** [comment]
Ah, I think this is because with pycobertura show coverage.xml the `.coveragerc` is probably not taken into account...

**Comment by gro1m** [comment]
And is not source actually a parameter:
https://github.com/aconrad/pycobertura/blob/master/pycobertura/cli.py#L92-L101

**Comment by gro1m** [comment]
@aconrad Ok, the problem is that we have a `setup.cfg` pytest section (in addition to `tox.ini`). It is recommended to have it in `tox.ini` (see also: https://docs.pytest.org/en/6.2.x/customize.html). Just removing it, unfortunately does not solve the problem...

**Comment by gro1m** [comment]
@aconrad Should be solved with: https://github.com/aconrad/pycobertura/pull/140. Note that apparently all coverage related stuff has to be in the `[testenv]` and not in the general `[pytest]` configuration section. That should solve the problem :)

**Comment by aconrad** [comment]
Oh awesome! Thanks! I'll test it on my machine tomorrow...

**Pull request merged**